subprojects/core/src/main/java/org/gradle/api/internal/TaskExecutionHistory.java

```diff
@@ -15,15 +15,16 @@
  */
 package org.gradle.api.internal;
 
-import org.gradle.api.file.FileCollection;
-
 import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Set;
 
 public interface TaskExecutionHistory {
     /**
      * Returns the set of output files which the task produced.
```

Comment:
I think we should clarify that we are also returning directories and missing files here now.